pkg/agent/manager/cache/cache.go

```diff
@@ -599,16 +604,21 @@ func (c *Cache) matchingIdentities(set selectorSet) []Identity {
 
 func (c *Cache) buildWorkloadUpdate(set selectorSet) *WorkloadUpdate {
 	w := &WorkloadUpdate{
-		Bundle:           c.bundles[c.trustDomainID],
-		FederatedBundles: make(map[string]*bundleutil.Bundle),
+		Bundle:           c.bundles[c.trustDomain],
+		FederatedBundles: make(map[spiffeid.TrustDomain]*bundleutil.Bundle),
 		Identities:       c.matchingIdentities(set),
 	}
 
 	// Add in the bundles the workload is federated with.
 	for _, identity := range w.Identities {
 		for _, federatesWith := range identity.Entry.FederatesWith {
-			if federatedBundle := c.bundles[federatesWith]; federatedBundle != nil {
-				w.FederatedBundles[federatesWith] = federatedBundle
+			td, err := spiffeid.TrustDomainFromString(federatesWith)
+			if err != nil {
+				c.log.WithField(telemetry.TrustDomainID, federatesWith).Warn("Federated Trust Domain invalid")
```

Comment:
We should add the error as a field:
```suggestion
				c.log.WithFields(logrus.Fields{
				    telemetry.TrustDomainID: id,
				    logrus.ErrorKey: err,
				}).Warn("Invalid federated trust domain")
```